openfpga_flow/tasks/fpga_bitstream/overwrite_bitstream/device_4x4/config/test.py

```diff
@@ -0,0 +1,119 @@
+import xml.etree.ElementTree as ET
+import sys
+import shutil
+import os
+import random
+
+random.seed()
+assert len(sys.argv) >= 2
+assert sys.argv[1] in ["run_golden", "generate_testcase", "validate"]
+TEST_BIT_COUNT = 200
+
+def read_fabric_bitstream_xml(file) :
+
+  bit_count = 0
+  tree = ET.parse(file)
+  root = tree.getroot()
+  assert root.tag == "fabric_bitstream", "Root tag is not 'fabric_bitstream', but '%s'" % root.tag
+  for region in root :
+    assert region.tag == "region", "fabric_bitstream child node tag is not 'region', but '%s'" % region.tag
+    for bit in region :
+      assert bit.tag == "bit", "region child node tag is not 'bit', but '%s'" % bit.tag
+      assert "path" in bit.attrib, "Attribute 'path' does not exist in bit node"
+      assert "value" in bit.attrib, "Attribute 'value' does not exist in bit node"
+      assert bit.attrib["value"] in ["0", "1"]
+      bit_count += 1
+  return [tree, bit_count]
+  
+def read_bitstream_annotation_xml(file) :
+  
+  xml = {}
+  tree = ET.parse(file)
+  root = tree.getroot()
+  assert root.tag == "openfpga_bitstream_setting", "Root tag is not 'openfpga_bitstream_setting', but '%s'" % root.tag
+  for overwrite_bitstream in root :
+    assert overwrite_bitstream.tag == "overwrite_bitstream", "openfpga_bitstream_setting child node tag is not 'overwrite_bitstream', but '%s'" % overwrite_bitstream.tag
+    for bit in overwrite_bitstream :
+      assert bit.tag == "bit", "overwrite_bitstream child node tag is not 'bit', but '%s'" % bit.tag
+      assert "path" in bit.attrib, "Attribute 'path' does not exist in bit node"
+      assert "value" in bit.attrib, "Attribute 'value' does not exist in bit node"
+      path = bit.attrib["path"]
+      assert path not in xml
+      index = path.rfind("[")
+      assert index != -1
+      path = "%s.mem_out%s" % (path[:index], path[index:])
+      assert path not in xml
+      assert bit.attrib["value"] in ["0", "1"]
+      xml[path] = bit.attrib["value"]
+  return xml
+
+if sys.argv[1] == "run_golden" :
+
+  openfpga_exe = os.path.abspath("../../../../../../../../../build/openfpga/openfpga")

```

Comment:
The test system sets macro "OPENFPGA_PATH" for all of its test tasks. You could use that setting instead of this abs path.